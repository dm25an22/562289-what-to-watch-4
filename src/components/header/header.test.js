import React from "react";
import renderer from "react-test-renderer";
import Header from "./header";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";

const mockStore = configureStore({});

const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`
};


it(`renders Headers component without authorization`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
it(`renders Headers component with authorization`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
