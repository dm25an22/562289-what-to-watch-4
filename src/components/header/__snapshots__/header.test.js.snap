// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Headers component with authorization 1`] = `
<header
  className="page-header undefined"
>
  <div
    className="logo"
  >
    <a
      className="logo__link undefined"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        src="/img/avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;

exports[`renders Headers component with class "movie-card__head" 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link undefined"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        src="/img/avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;

exports[`renders Headers component with class "user-page__head" and children 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link undefined"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    My list
  </h1>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        src="/img/avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;

exports[`renders Headers component without authorization 1`] = `
<header
  className="page-header undefined"
>
  <div
    className="logo"
  >
    <a
      className="logo__link undefined"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign in
    </a>
  </div>
</header>
`;
